src/multiselect.js

```diff
@@ -190,6 +192,20 @@ export class Multiselect {
     // on Block Selected
     if (e.type === Blockly.Events.SELECTED) {
       this.controls_.updateMultiselect();
+    // on Block field changed
+    } else if (!this.isUpdatingBlocks_ && e.type === Blockly.Events.CHANGE &&
+        e.element === 'field' && this.blockSelection_.has(e.blockId)) {
+      this.isUpdatingBlocks_ = true;
+      const blockType = this.workspace_.getBlockById(e.blockId).type;
+      // Update the fields to the same value for
+      // the selected blocks with same type.
+      this.blockSelection_.forEach((id) => {
+        const block = this.workspace_.getBlockById(id);
+        if (block.type === blockType) {
+          block.setFieldValue(e.newValue, e.name);
+        }
+      });
+      this.isUpdatingBlocks_ = false;
```

Comment:
I feel like this should start a group if one doesn't exist. Otherwise, if you change multiple blocks at the same time then deselect and hit Ctrl+Z you have to step through each block that changed rather than reverting the changes as a single event.

Reply:
Just fixed that and make sure each block return to their original state when undoing:
![GIF 02-08-2022 10-31-35](https://user-images.githubusercontent.com/43995067/182279130-83f6ca89-eb46-4e0a-98eb-8082b5549daf.gif)